common/scala/src/main/scala/whisk/core/containerpool/logging/DockerLogStore.scala

```diff
@@ -43,7 +69,22 @@ class DockerLogStore(system: ActorSystem) extends LogStore {
   override def collectLogs(transid: TransactionId,
                            container: Container,
                            action: ExecutableWhiskAction): Future[ActivationLogs] = {
-    container.logs(action.limits.logs.asMegaBytes, action.exec.sentinelledLogs)(transid).map(ActivationLogs(_))
+
+    val possibleErrors = Set(Messages.logFailure, Messages.truncateLogs(action.limits.logs.asMegaBytes))
+
+    container
+      .logs(action.limits.logs.asMegaBytes, action.exec.sentinelledLogs)(transid)
+      .via(DockerLogStore.toFormattedString)
+      .runWith(Sink.seq)
+      .flatMap { seq =>
+        val errored = seq.lastOption.exists(last => possibleErrors.exists(last.contains))
+        val logs = ActivationLogs(seq.toVector)
+        if (!errored) {
+          Future.successful(logs)
```

Comment:
nit: the future here seems superfluous; this could have been an `either` type.

Reply:
Note the function is already returning a Future, this just makes it a bit more granular.